### PR TITLE
Add styling to guidance tab

### PR DESCRIPTION
**Pull request by iHiD**
@kntsoriano This adds styling to the guidance tab. It's going to need a little bit of work from you. Take a read of the comments - I've added TODOs for the various bits. I think it should be a quick one 🙂 

There is also a bug/missing-feature. Clicking an Accordion Heading that is open should close it (so that all the sections become closed). Could you fix that on this PR too please?

### CODE REVIEW

File: app/helpers/react_components/mentoring/discussion.rb [hunk 1]
```diff
@@ -9,6 +9,10 @@ def to_s
           {
             discussion_id: discussion.uuid,
             user_id: current_user.id,
+            mentor: {
+              handle: mentor.handle,
+              avatar_url: mentor.avatar_url
+            },
```
Comment (iHiD) [inline app/helpers/react_components/mentoring/discussion.rb]: I've added this new key for displaying the info of the mentor. You could add these as prefixed keys (e.g. `mentor_handle`) into the `mentor_solution` block instead if you prefered that. I don't have a huge preference. I put them here as I figured we may choose to use them elsewhere in the future too.

File: app/helpers/react_components/mentoring/discussion.rb [hunk 2]
```diff
@@ -72,21 +77,22 @@ def iterations
...
-      def track
-        discussion.solution.track
+        {
+          snippet: ms.snippet,
+          num_loc: ms.num_loc,
+          num_stars: ms.num_stars,
+          num_comments: ms.num_comments,
+          published_at: ms.published_at,
+          web_url: Exercism::Routes.private_solution_url(ms)
+        }
```
Comment (iHiD) [inline app/helpers/react_components/mentoring/discussion.rb:93]: These values are all correct and don't need you to do anything.

File: app/javascript/components/common/Accordion.tsx [hunk 3]
```diff
@@ -45,7 +46,12 @@ const AccordionHeader = ({
       aria-controls={`accordion-panel-${id}`}
       id={`accordion-header-${id}`}
     >
-      {children}
+      {isOpen ? (
+        <GraphicalIcon icon="minus-circle" />
+      ) : (
+        <GraphicalIcon icon="plus-circle" />
+      )}
```
Comment (iHiD) [inline app/javascript/components/common/Accordion.tsx:53]: Maybe you could add a test for this please?

File: app/helpers/react_components/mentoring/discussion.rb [hunk 4]
```diff
@@ -72,21 +78,40 @@ def iterations
...
+    <code>each_cons</code> instead of an iterator
+    <code>with_index</code>: In Ruby, you rarely have to write
+    iterators that need to keep track of the index. Enumerable has
+    powerful methods that do that for us.
+  </li>
+  <li>
+    <code>chars</code>: instead of <code>split("")</code>.
+  </li>
+</ul>
+        '.strip
```
Comment (iHiD) [inline app/helpers/react_components/mentoring/discussion.rb:111]: I will do this in a seperate commit/PR. It's not blocking this.

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 5]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
   )
 
   return (
-    <div>
+    <>
+      <Accordion id="notes" isOpen={isOpen('notes')} onClick={handleClick}>
+        <Accordion.Header>Mentor notes</Accordion.Header>
+        <Accordion.Panel>
+          <div className="c-textual-content --small">
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

If notes is `present?` then this should show. There will also be a CTA to improve the notes. 

If notes is not present, then we'll show a (different) CTA to contribute instead.

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 6]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+            {/* TODO: Replace with `notes` */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 7]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
         <Accordion.Header>How you solved the exercise</Accordion.Header>
         <Accordion.Panel>
-          <p>Solution here</p>
-        </Accordion.Panel>
-      </Accordion>
-      <Accordion id="notes" isOpen={isOpen('notes')} onClick={handleClick}>
-        <Accordion.Header>Mentor notes</Accordion.Header>
-        <Accordion.Panel>
-          <p>Notes here</p>
+          {/* TOOD: Replace "#" with mentor_solution.web_url */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

- There are various things that need the proper data putting in here.
- If mentor_solution is `null` then this whole accordion section (including the AccordionHeader) should not render.

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 8]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+          <a href="#" className="c-published-solution">
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: Note, that there is a normal published-solution component, but has slightly different data from this so I don't think we want to use `Common` component here (at least for now)

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 9]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+                      alt="Number of times solution has been commented on"
+                    />
+                    {/*TODO: Replace with num_comments */}
+                    <div className="--num">2</div>
+                  </div>
+                </div>
+              </div>
+            </header>
+            <pre>
+              {/* TODO: Replcae with mentor_solution.snippet */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 10]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+                  <span className="hljs-keyword">return</span>{' '}
+                  <span className="hljs-literal">true</span>
+                </span>
+                ;
+              </code>
+            </pre>
+            <footer className="--footer">
+              <div className="locs">
+                <GraphicalIcon icon="loc" />
+                {/* TODO: Replace with mentor_solution.num_loc */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 11]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+                    />
+                    {/*TODO: Replace with num_stars */}
+                    <div className="--num">10</div>
+                  </div>
+                  <div className="--count">
+                    <Icon
+                      icon="comment"
+                      alt="Number of times solution has been commented on"
+                    />
+                    {/*TODO: Replace with num_comments */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 12]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+                </div>
+
+                {/*TODO: If mentor_solution.published_at? */}
+                <div className="--counts">
+                  <div className="--count">
+                    <Icon
+                      icon="star"
+                      alt="Number of times solution has been stared"
+                    />
+                    {/*TODO: Replace with num_stars */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 13]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+                  src="https://avatars2.githubusercontent.com/u/5337876?s=460&v=4"
+                />
+
+                <div className="--info">
+                  <div className="--exercise-title">Your Solution</div>
+                  {/* TODO: Replace "Leap" and "Ruby" */}
+                  <div className="--track-title">to Leap in Ruby</div>
+                </div>
+
+                {/*TODO: If mentor_solution.published_at? */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 14]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
+              <div className="--exercise">
+                {/* TODO: Replace with mentor.handle and mentor.avatar_url */}
+                <Avatar
+                  handle="iHiD"
+                  src="https://avatars2.githubusercontent.com/u/5337876?s=460&v=4"
+                />
+
+                <div className="--info">
+                  <div className="--exercise-title">Your Solution</div>
+                  {/* TODO: Replace "Leap" and "Ruby" */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/mentoring/discussion/Guidance.tsx [hunk 15]
```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
...
-      </Accordion>
-      <Accordion id="notes" isOpen={isOpen('notes')} onClick={handleClick}>
-        <Accordion.Header>Mentor notes</Accordion.Header>
-        <Accordion.Panel>
-          <p>Notes here</p>
+          {/* TOOD: Replace "#" with mentor_solution.web_url */}
+          <a href="#" className="c-published-solution">
+            <header className="--header">
+              <div className="--exercise">
+                {/* TODO: Replace with mentor.handle and mentor.avatar_url */}
```
Comment (iHiD) [inline app/javascript/components/mentoring/discussion/Guidance.tsx]: **TODO for you.**

File: app/javascript/components/common/Accordion.tsx (hunk 3 above)
Comment (kntsoriano) [reply app/javascript/components/common/Accordion.tsx:53]: I think this is fine. We already have a test for the correct aria attributes in the JS. We can't also assert on the tests unless we change this to an `<Icon />`and add the correct alt attribute. The alt attribute might be redundant as we already have supplied the correct aria attributes.

**Comment by kntsoriano** [comment]
@iHiD Added the commits.

File: app/helpers/react_components/mentoring/discussion.rb [hunk 16]
```diff
@@ -72,21 +74,45 @@ def iterations
         end
       end
 
-      memoize
-      delegate :student, to: :discussion
+      def mentor_solution
+        ms = Solution.for(mentor, exercise)
+        return nil unless ms
```
Comment (ErikSchierboom) [inline app/helpers/react_components/mentoring/discussion.rb:79]: I see a lot of places where `return unless` is used, without the explicit `nil`. Is there a reason when to add it or not?

File: app/javascript/components/mentoring/discussion/MentorNotes.tsx [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+export const MentorNotes = ({ notes }: { notes?: string }): JSX.Element => {
+  if (notes) {
+    return (
+      <React.Fragment>
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/discussion/MentorNotes.tsx:6]: I'm seeing usage of both `<>` and `React.Fragment` in various places. They are the same, right? And if so, what is the benefit of using `React.Fragment` over `<>`?

**Review by ErikSchierboom — APPROVED** [review]

File: app/helpers/react_components/mentoring/discussion.rb (hunk 16 above)
Comment (iHiD) [reply app/helpers/react_components/mentoring/discussion.rb:79]: I never remember that `return unless ms` returns `nil` so I always add it. I quite like the explicitness. But the two are equivalent. I'm surprised Rubocop doesn't standardise.

File: app/javascript/components/mentoring/discussion/MentorNotes.tsx (hunk 17 above)
Comment (iHiD) [reply app/javascript/components/mentoring/discussion/MentorNotes.tsx:6]: I always use `<>` if I can, but if attributes are needed (for example a `ref="..."` then you have to use the full `<React.Fragment>`. 

@kntsoriano may know of some other reasons that I'm not aware of.

**Comment by iHiD** [comment]
Sweet work @kntsoriano. I'm merging this down! Thanks for the review @ErikSchierboom :)

File: app/javascript/components/mentoring/discussion/MentorNotes.tsx (hunk 17 above)
Comment (kntsoriano) [reply app/javascript/components/mentoring/discussion/MentorNotes.tsx:6]: I don't know why I choose one over the other. I guess doing `<React.Fragment />`is the wiser choice as we don't need to change much if we want to add ref to it? No spectific preference, though.

**Pull request merged**